sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala

```diff
@@ -39,7 +39,7 @@ class FileStreamSource(
     providerName: String,
     dataFrameBuilder: Array[String] => DataFrame) extends Source with Logging {
 
-  private val fs = FileSystem.get(sqlContext.sparkContext.hadoopConfiguration)
+  private val fs = new Path(path).getFileSystem(sqlContext.sparkContext.hadoopConfiguration)
```

Comment:
All LGTM pending tests. I suppose this could even be fetched in the one place it's used later during the method call rather than hold on to a reference, but, I can't recall a specific reason it's bad to hold onto a FileSystem handle, so, leave it as you've done I think.